### PR TITLE
Generate a service account token to make it work in 1.24

### PR DESCRIPTION
**Pull request by futuretea**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

Versions of Kubernetes before v1.22 automatically created credentials for accessing the Kubernetes API. This older mechanism was based on creating token Secrets that could then be mounted into running Pods. In more recent versions, including Kubernetes v1.25, API credentials are obtained directly by using the [TokenRequest](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/) API

**Solution:**

https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-service-account-api-token

**Related Issue:**
https://github.com/harvester/harvester/issues/2889

**Test plan:**

### CODE REVIEW

File: pkg/api/kubeconfig/generate_handler.go [hunk 1]
```diff
@@ -197,6 +197,35 @@ func (h *GenerateHandler) ensureSaAndSecret(namespace, name string) (*corev1.Ser
...
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: rbacv1.SchemeGroupVersion.Version,
+						Kind:       rbacv1.ServiceAccountKind,
+						Name:       sa.Name,
+						UID:        sa.UID,
+					},
+				},
+				Annotations: map[string]string{
+					"kubernetes.io/service-account.name": sa.Name,
```
Comment (guangbochen) [inline pkg/api/kubeconfig/generate_handler.go]: for the key prefer to use `v1.ServiceAccountNameKey` instead and add `v1.ServiceAccountUIDKey` as well.

Comment (futuretea) [reply pkg/api/kubeconfig/generate_handler.go]: updated

**Review by yaocw2020 — APPROVED** [review]
LGTM, thanks.

**Review by guangbochen — APPROVED** [review]
LGTM, and verified it works well with Rancher v2.6.9-rc3.
image

**Pull request merged**